docs/src/main/asciidoc/amqp-reference.adoc

```diff
@@ -0,0 +1,647 @@
+    .addKeyPath("./tls/tls.key");
+  PemTrustOptions trust = new PemTrustOptions().addCertPath("./tlc/ca.crt");
+  return new AmqpClientOptions()
+        .setSsl(true)
+        .setPemKeyCertOptions(keycert)
+        .setPemTrustOptions(trust)
+        .addEnabledSaslMechanism("EXTERNAL")
+        .setHostnameVerificationAlgorithm("")
+        .setConnectTimeout(30000)
+        .setReconnectInterval(5000)
+        .setContainerId("my-container");
+}
+----
+
+This instance is retrieved and used to configure the client used by the connector.
+You need to indicate the name of the client using the `client-options-name` attribute:
+
+[source, properties]
+----
+mp.messaging.incoming.prices.client-options-name=my-named-options
+----
+
+== Health reporting
+
+If you use the AMQP connector with the `quarkus-smallrye-health` extension, it contributes to the readiness and liveness probes.
+The AMQP connector reports the readiness and liveness of each channel managed by the connector.
+At the moment, the AMQP connector uses the same logic for the readiness and liveness checks.
+
+To disable health reporting, set the health-enabled attribute for the channel to false.
+On the inbound side (receiving messages from AMQP), the check verifies that the receiver is attached to the broker.
+On the outbound side (sending records to AMQP), the check verifies that the sender is attached to the broker.
+
+Note that a message processing failures nacks the message, which is then handled by the failure-strategy.
+It the responsibility of the `failure-strategy` to report the failure and influence the outcome of the checks.
+The `fail` failure strategy reports the failure, and so the check will report the fault.
+
+== Using RabbitMQ
+This connector is for AMQP 1.0. RabbitMQ implements AMQP 0.9.1.
+RabbitMQ does not provide AMQP 1.0 by default, but there is a plugin for it.
+To use RabbitMQ with this connector, enable and configure the AMQP 1.0 plugin.
+
+Despite the plugin, a few features won’t work with RabbitMQ.
+Thus, we recommend the following configurations.
+
+To receive messages from RabbitMQ:
+
+* Set durable to false
+
+[source, properties]
+----
+mp.messaging.incoming.prices.connector=smallrye-amqp
+mp.messaging.incoming.prices.durable=false
+----
+
+To send messages to RabbitMQ:
+
+* set the destination address (anonymous sender are not supported)
+* set use-anonymous-sender to false
+
+[source, properties]
+----
+mp.messaging.outgoing.generated-price.connector=smallrye-amqp
+mp.messaging.outgoing.generated-price.address=prices
+mp.messaging.outgoing.generated-price.use-anonymous-sender=false
+----
+
+As a consequence, it’s not possible to change the destination dynamically (using message metadata) when using RabbitMQ.
+
+[[configuration-reference]]
+== AMQP Connector Configuration Reference
+
+=== Quarkus specific configuration
+
+include::{generated-dir}/config/quarkus-smallrye-reactive-messaing-amqp.adoc[opts=optional, leveloffset=+1]
```

Comment:
Hmmmm, `messaing`? Either the include is incorrect or something wrong in the config definition.

Reply:
typo.

Reply:
The include was still incorrect. I fixed it in a separate commit together with a few other minor issues.